internal/coss/license/license.go

```diff
@@ -195,7 +196,10 @@ func (lm *ManagerImpl) Shutdown(ctx context.Context) error {
 			return err
 		}
 		// deactivate the license for this machine so it can be used on another machine
-		if err := lm.license.Deactivate(ctx, fingerprint); err != nil {
+		// Use a dedicated timeout context for deactivation to avoid competing with other shutdown operations
+		deactivateCtx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```

Comment:
I'd prefer if this new context was defined in `main.go` and then passed into `licenseManagerShutdown(...)` instead of `shutdownCtx`.